src/collectives/device/msccl_kernel_impl.h

```diff
@@ -372,7 +372,7 @@ __device__ __forceinline__ void mscclRunInterpreter(
           return;
       }
       if (t->hasDependence && tid == nthreads-1)
-        __atomic_store_n(&mscclFlags[bid].flag, (uint64_t) COMPUTE_FLAG(workIndex, iter, step), t->type == MSCCL_REDUCE ? __ATOMIC_RELEASE : __ATOMIC_RELAXED);
+        __atomic_store_n(&mscclFlags[bid].flag, (uint64_t) COMPUTE_FLAG(workIndex, iter, step), (t->type == MSCCL_REDUCE || t->type == MSCCL_RECV) ? __ATOMIC_RELEASE : __ATOMIC_RELAXED);
```

Comment:
Do `recv` operations have any dependence for any collectives?

Reply:
"recv" can be receive from remote GPU send or RDMA send. It appears that RDMA path needs memory coherence atomic. However, the 2 paths can not be easily distinguished within MSCCL kernel. Thus I added atomic to both.

Reply:
That's alright. But my concern is this `if` statement is executed only when the operation depends on another operation (`t->hasDependence`). Do `recv` operations have dependence on any other operations on single/multi node?